core/src/main/java/feign/Response.java

```diff
@@ -242,4 +246,19 @@ public String toString() {
       return decodeOrDefault(data, UTF_8, "Binary data");
     }
   }
+
+  private static Map<String, Collection<String>> caseInsensitiveCopyOf(Map<String, Collection<String>> headers) {
```

Comment:
I'm thinking of leaving this at is now. Testing it this way seems a bit cumbersome, but it's safer.